src/boundingmesh/Mesh.cpp

```diff
@@ -189,367 +189,44 @@ void swap(Mesh& first, Mesh& second) {
-  int state = 0;
-  unsigned int n_vertices = 0;
-  unsigned int n_faces = 0;
-  unsigned int n_edges = 0;
-  while (!file.eof() && file.good()) {
-    std::getline(file, line);
-    line.erase(0, line.find_first_not_of(' '));
-    line.erase(line.find_last_not_of(" \n\r\t") + 1);
-    if (line.size() == 0) continue;
-    if (line.at(0) == '#') continue;
-
-    std::stringstream sstream(line);
-    Real x, y, z;
-    Index a, b, c;
-
-    int n_face_vertices;
-
-    switch (state) {
-      case 0:  // Reading Preamble
-        if (line == "OFF")
-          state = 1;
-        else {
-          std::cout << "Error reading .off file: first line is bad:"
-                    << line.length() << (int)line.at(3) << std::endl;
-          return;
-        }
-        break;
-      case 1:  // Reading vertex/face count
-        sstream >> n_vertices >> n_faces >> n_edges;
-        if (n_vertices == 0 || n_faces == 0) {
-          std::cout << "Error reading .off file: Vertex/face count is zero"
-                    << std::endl;
-          return;
-        }
-        state = 2;
-        break;
-      case 2:                    // Reading vertices
-        sstream >> x >> y >> z;  // Only read position, ignore color if existent
-        addVertex(Vector3(x, y, z));
-
-        if (vertices_.size() == n_vertices) state = 3;
-        break;
-      case 3:  // Reading faces
-        sstream >> n_face_vertices;
-        if (n_face_vertices < 3) {
-          std::cerr
-              << "Error reading .off file: Reading face with 2 or less vertices"
-              << std::endl;
-          clean();
-          return;
-        } else if (n_face_vertices == 3) {
-          sstream >> a >> b >>
-              c;  // Only read geometric information, ignore color if existent
-          if (a > vertices_.size() || b > vertices_.size() ||
-              c > vertices_.size())
-            std::cout << "Error reading .off file: invalid index" << std::endl;
-          addTriangle(a, b, c);
-        } else {
-          std::cout << "Error reading .off file: Reading face with more than 3 "
-                       "vertices"
-                    << std::endl;
-        }
-        break;
-      default:
-        std::cout << "Error reading .off file: Broken reading state"
-                  << std::endl;
-        clean();
-        return;
-    }
-  }
-  n_original = vertices_.size();
-  if (vertices_.size() != n_vertices || triangles_.size() != n_faces) {
-    std::cerr << "Error reading .off file: Vertex/face count is incorrect"
-              << std::endl;
-    clean();
+void Mesh::loadFile( const std::string& pFile)
+{
+  // Create an instance of the Importer class
+  Assimp::Importer importer;
+  // And have it read the given file with some example postprocessing
+  // Usually - if speed is not the most important aspect for you - you'll 
+  // propably to request more postprocessing than we do in this example.
+  const aiScene* scene = importer.ReadFile( pFile, 
```

Comment:
This is quite a change.
Please confirm you are actually planning fix bugs if this introduces any.

Reply:
I will